src/parseFromAnchor.js

```diff
@@ -12,7 +26,7 @@ function parseFromAnchor(opt_uri) {
 		hostname: link.hostname,
 		password: link.password,
 		pathname: link.pathname[0] === '/' ? link.pathname : '/' + link.pathname,
-		port: link.port,
+		port: extractPortFromURI(opt_uri),
```

Comment:
This will be only used for the test. If the user chooses port 0 it will work. This test only detects if the browser natively behaves wrongly.